core/src/main/resources/lib/hudson/propertyTable.jelly

```diff
@@ -63,14 +66,20 @@ THE SOFTWARE.
           <td style="white-space: normal" class="property-table__value">
             <j:choose>
               <j:when test="${attrs.sensitive}">
-                <div class="hidden-info">
-                  <div class="jenkins-button">
-                    ${%reveal}
-                  </div>
+                <div class="app-hidden-info-reveal">
+                  <button class="jenkins-button jenkins-button--transparent">
+                    <span class="jenkins-!-color-blue"><l:icon src="symbol-view"/> ${%reveal}</span>
+                  </button>
+                </div>
+                <div class="app-hidden-info-hide jenkins-hidden">
+                  <span class="sensitive">
+                    <button class="jenkins-button jenkins-button--transparent">
+                      <span class="jenkins-!-color-blue">
+                        <l:icon src="symbol-eye-off-outline"/><j:out value=" "/><l:breakable value="${e.value}"/>
```

Comment:
Just a thought - would inverting the icons make more sense? So when hidden, you've got 'symbol-eye-off-outline' and when visible it's the other icon?

Reply:
I think this is the right way round, when it’s hidden you click the eye to open it up